ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import android.content.Context
+import com.revenuecat.purchases.Package
+import java.text.NumberFormat
+import java.util.Currency
+import java.util.Locale
+
+@Suppress("UnusedParameter", "FunctionOnlyReturningConstant")
+internal object VariableDataProvider {
+    private const val MICRO_MULTIPLIER = 1000000.0
+
+    fun applicationName(context: Context): String {
+        return context.applicationInfo.loadLabel(context.packageManager).toString()
+    }
+
+    fun localizedPrice(rcPackage: Package): String {
+        return rcPackage.product.price.formatted
+    }
+
+    fun localizedPricePerMonth(rcPackage: Package, locale: Locale): String {
+        val price = rcPackage.product.price.amountMicros / MICRO_MULTIPLIER
+        val periodMonths = rcPackage.product.period?.valueInMonths ?: 1.0
+        val currencyCode = rcPackage.product.price.currencyCode
+        val numberFormat = NumberFormat.getCurrencyInstance(locale)
+        numberFormat.currency = Currency.getInstance(currencyCode)
+        return numberFormat.format(price / periodMonths)
```

Comment:
What do you think about putting this in `purchases` instead? 
This also needs tests to ensure rounding is correct for the different types of package (see `SubscriptionPeriodTests` in iOS)`.

Reply:
I thought about it and wasn't sure due to:
- This is mostly an approximation. We assume a few things in calculating the number of months per period (Like that the month has 30 days, or 4 weeks).
- We may need to add an optional locale parameter and need to recompute this everytime, otherwise, if the locale changes, the value may change
- We would need to add this function both in `PricingPhase` and `StoreProduct`, with slightly different implementations for Amazon and Google.
- I haven't tested this too much yet, so I would like some more testing.

Having said that, I do think this could still be extracted to `purchases` for convenience. But clearly that deserves its own PR.

Reply:
In the end, I think it should be fine to move it there. Did it in #1255